src/py/pyodide/http.py

```diff
@@ -270,6 +328,16 @@ async def unpack_archive(
         filename = self._url.rsplit("/", -1)[-1]
         unpack_buffer(buf, filename=filename, format=format, extract_dir=extract_dir)
 
+    def abort(self, reason: Any = None) -> None:
```

Comment:
I just realized that we can only call this `abort` method after the fetch is done. Is it a useful behavior? Can it be a reason for why node tests are failing (sending abort signal after the fetch is finished)?

Reply:
Oh, never mind. This raises AbortError in both Chrome and Node, so sending abort signal after the fetch seems fine.

```js
let controller = new AbortController();

let response = await fetch('http://example.com', {
    signal: controller.signal
});

controller.abort("reason")
await response.text()
```

Reply:
tldr: It may be the reason why `test_pyfetch_manually_abort` failed, but I think it is why `test_pyfetch_custom_abort_signal` failed.

> we can only call this `abort` method after the fetch is done. Is it useful?

IIUC, a fetch in js need to be await twice, first in `await fetch(...`, the second in `await resp.text()`. In python, cancelling a future, or using a custom `AbortController` passed in `kwargs` can abort in both `await`, and they can use `FetchResponse.abort` in the second stage.

The only way to stop a fetch (or a .text()/.json() method of a fetch's response) is to call the `abort` method on the `AbortController` instance. And I added this `FetchResponse.abort()` method just for convenience. Because I personally think python developers may not want to create a `AbortController` and abort it themselves, so I expose a abort method for them to stop the request.

But to be honest, it is only useful in the second stage (await resp.text()), so it may not be very useful and may cause confusion. If you think the drawbacks outweight the benefits I agree to remove that.

In `test_pyfetch_manually_abort`, I abort the resposne after the first stage. Maybe the request is already ended at that time? In other words, maybe the second stage is already ended before I called `resp.abort`.

But in `test_pyfetch_custom_abort_signal`, I abort the signal before calling `pyfetch`, so the request should never start. I can't think of a reason why this happens. I'm trying to reproduce this though.

Reply:
> But to be honest, it is only useful in the second stage, so it may not be very useful and may cause inconsistency and confusion. If you think the drawbacks outweight the benefits I agree to remove that.

No, I think it is fine to keep the `abort` method in `FetchResponse`. Even if there are not many real-world use cases, someone may use it, and using it for testing isn't that bad.

We may add another Python helper class or function to abort the "first" stage of fetch too as a follow-up PR if you think it might be useful for people.